Ensure promise is not garbage collected

stwrt opened the pull request:
#### c29e5a28c1760dd0bef2da987ab11eb7e8ba50ee
<pre>
Ensure promise is not garbage collected
<a href="https://bugs.webkit.org/show_bug.cgi?id=242287">https://bugs.webkit.org/show_bug.cgi?id=242287</a>

Reviewed by Tim Nguyen.

We need to ensure that the promise always remains alive when in use.
Adding a RefPtr guarantees that it will not be garbage collected.

* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::unwrapKey):

Canonical link: <a href="https://commits.webkit.org/252091@main">https://commits.webkit.org/252091@main</a>
</pre>

nt1m commented on Source/WebCore/crypto/SubtleCrypto.cpp:
```diff
@@ -1124,7 +1124,7 @@ void SubtleCrypto::unwrapKey(JSC::JSGlobalObject& state, KeyFormat format, Buffe
     auto callback = [index, weakThis, format, importAlgorithm, unwrappedKeyAlgorithm = crossThreadCopyImportParams(*unwrappedKeyAlgorithm), extractable, keyUsagesBitmap](const Vector<uint8_t>& bytes) mutable {
         if (weakThis) {
-            if (auto promise = weakThis->m_pendingPromises.get(index)) {
+            if (RefPtr<DeferredPromise> promise = weakThis->m_pendingPromises.get(index)) {
```
```suggestion
            if (RefPtr promise = weakThis->m_pendingPromises.get(index)) {
```

I *think* this should work fine

stwrt replied:
just tested it and it works 👍

nt1m reviewed (approved):
Took me a bit to understand why `HashMap::get()` was returning a raw pointer, when `m_pendingPromises` has `Ref<>` as values. For posterity, it's because `HashMap::get()` returns a `PeekType` which corresponds to a raw pointer (see `using PeekType = P*;`).

stwrt committed:
Ensure promise is not garbage collected
https://bugs.webkit.org/show_bug.cgi?id=242287

Reviewed by Tim Nguyen.

We need to ensure that the promise always remains alive when in use.
Adding a RefPtr guarantees that it will not be garbage collected.

* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::SubtleCrypto::unwrapKey):

Canonical link: https://commits.webkit.org/252091@main
